### ISSUE TITLE
`TrailingCommaOnDeclarationSite` shows incorrect line number and wrong error message for enums

### ISSUE THREAD

**Issue by igorwojda** [pinterest/ktlint#1827]
## Expected Behavior

Display the correct line number and error message.

## Observed Behavior

The incorrect line number is displayed together with the irrelevant error message
```
* What went wrong:
Execution failed for task ':spotlessKotlin'.
> Error on line: 1, column: 1
  rule: trailing-comma-on-declaration-site
  Missing trailing comma before "}"

```

## Steps to Reproduce

```kotlin
enum class Priority(private val value: Int) {
    PRIORITY_1(1),
    PRIORITY_2(2);

    companion object {
        fun getByValue(value: Int): Priority {
            return Priority.values().firstOrNull {
                it.value == value
            } ?: throw IllegalArgumentException("Invalid priority value: $value ")
        }
    }
}

```

## Your Environment

* Version of ktlint used: `0.48.2`
* Name and version (or code for the custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `./gradlew spotlessCheck`
* Version of Gradle used (if applicable): `8.0`
* Operating System and version: `MacOS Ventura`

**Comment by paul-dingemans** [comment]
This does not seem to be a problem in Ktlint. With Ktlint CLI I can not reproduce this problem:
```
$ cat <<EOF | ktlint-0.48.2 --stdin
> enum class Priority(private val value: Int) {
>     PRIORITY_1(1),
>     PRIORITY_2(2);
> 
>     companion object {
>         fun getByValue(value: Int): Priority {
>             return Priority.values().firstOrNull {
>                 it.value == value
>             } ?: throw IllegalArgumentException("Invalid priority value: $value ")
>         }
>     }
> }
> EOF
21:04:35.087 [main] INFO com.pinterest.ktlint.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
<stdin>:3:18: Missing trailing comma before ";" (trailing-comma-on-declaration-site)

Summary error count (descending) by rule:
  trailing-comma-on-declaration-site: 1
```

Can you check with Ktlint CLI directly instead of using Spotless?

**Comment by m15o** [comment]
I have the same problem, and I could reproduce with Ktlint CLI using `--format` option:

```bash
cat <<EOF | ktlint --format --stdin
enum class Priority(private val value: Int) {
    PRIORITY_1(1),
    PRIORITY_2(2);

    companion object {
        fun getByValue(value: Int): Priority {
            return Priority.values().firstOrNull {
                it.value == value
            } ?: throw IllegalArgumentException("Invalid priority value: $value ")
        }
    }
}
EOF
```

```
09:55:00.513 [main] INFO com.pinterest.ktlint.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
enum class Priority(private val value: Int) {
    PRIORITY_1(1),
    PRIORITY_2(2),
    ;

    companion object {
        fun getByValue(value: Int): Priority {
            return Priority.values().firstOrNull {
                it.value == value
            } ?: throw IllegalArgumentException("Invalid priority value:  ")
        }
    }
}
<stdin>:1:1: Missing trailing comma before "}" (trailing-comma-on-declaration-site)
<stdin>:5:4: Unnecessary semicolon (no-semi)

Summary error count (descending) by rule:
  no-semi: 1
  trailing-comma-on-declaration-site: 1
```

**Comment by paul-dingemans** [comment]
Tnx. The `--format` was indeed the missing link to reproduce.

**Comment by paul-dingemans** [comment]
I can reproduce the problem with the `0.48.2` version. However, with current snapshot for the upcoming `0.49.x`, I can no longer reproduce it.
